apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/MetricRegistryReporter.java

```diff
@@ -67,8 +68,9 @@ public void stop() throws Exception {
     @Override
     public void report(Map<? extends Labels, MetricSet> metricSets) {
         if (tracer.isRunning()) {
+            List<String> serviceNames = tracer.getServiceNameOverrides();
```

Comment:
If the service name is explicitly set, should the metrics be reported for that service name?

Reply:
Yes, multiple service names are not supported when setting an explicit `service_name` in the config. See also `co.elastic.apm.agent.impl.ElasticApmTracerTest#testNotOverrideServiceNameWhenDefaultServiceNameConfigured`

Reply:
You can also just omit the service name so that the global one is used.

Reply:
Nevermind, that's what you did anyway.